src/ucp/am/eager.c

```diff
@@ -217,3 +217,141 @@ static ucp_proto_t ucp_eager_bcopy_single_proto = {
     .abort    = ucp_request_complete_send
 };
 UCP_PROTO_REGISTER(&ucp_eager_bcopy_single_proto);
+
+static ucs_status_t
+ucp_proto_eager_am_zcopy_single_init(const ucp_proto_init_params_t *init_params)
+{
+    ucp_context_t *context                = init_params->worker->context;
+    ucp_proto_single_init_params_t params = {
+        .super.super         = *init_params,
+        .super.latency       = 0,
+        .super.overhead      = 0,
+        .super.cfg_thresh    = context->config.ext.zcopy_thresh,
+        .super.cfg_priority  = 30,
+        .super.min_length    = 0,
+        .super.max_length    = SIZE_MAX,
+        .super.min_frag_offs = ucs_offsetof(uct_iface_attr_t, cap.am.min_zcopy),
+        .super.max_frag_offs = ucs_offsetof(uct_iface_attr_t, cap.am.max_zcopy),
+        .super.max_iov_offs  = UCP_PROTO_COMMON_OFFSET_INVALID,
```

Comment:
btw, @yosefe are we going to extend single zcopy protocols to work with multiple iovs?
what is the reason for this design decision (the logic is the same with TAGs)?

Reply:
single protocol can't support iov since in the general case it may need to send multiple messages (long iov list)
the reason for separating is that single can be more optimized for the common case of contig data type with small-medium message size

Reply:
in theory it may support iovs up to the certain limit, like it is done in proto v1, but ic the point, thanks

Reply:
I guess so, we can probably have a more fine-grained selection and use # of IOVs to decide. it will have slightly more overhead than contig, though.